libs/execution_context/ebpf_link.c

```diff
@@ -11,18 +11,25 @@
 
 /**
  * @brief State of the link between program and provider.
- * The link starts in IDLE state. When a program is attached to a provider, the
- * state transitions to ATTACHING. If the provider is not found, the state
- * transitions to IDLE. If the provider is found, the state transitions to
- * ATTACHED. When a program is detached from a provider, the state transitions
- * to DETACHING. Once the provider is notified, the state transitions to IDLE.
+ * Legal state transitions are:
+ * 1. EBPF_LINK_STATE_INITIAL -> EBPF_LINK_STATE_ATTACHING - In ebpf_link_attach_program when the program is being
```

Comment:
Do we really want to mention the specific (current) _functions_ where this transition is made?  Future code changes _could_ make such comments incorrect and thus confusing. IMO, just the part of the comment describing the transition ought to be enough.  Thoughts?